.github/workflows/appchain-mac.yml

```diff
@@ -0,0 +1,33 @@
+name: Task - Appchain Mac
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  appchain_mac:
+    strategy:
+      fail-fast: false
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - uses: ./.github/actions/setup-mac
+
+      - name: Setup app chain and run tests
+        run: |
+          cargo run -p madara create app-chain &
+          MADARA_PID=$!
+          until [ "$(docker inspect -f '{{.State.Running}}' bootstrapper_l2 2>/dev/null)" = "true" ]; do
+            echo "Waiting for Bootstrapper L2 container to start..."
+            sleep 1
+          done
+          echo "Waiting for Bootstrapper L2 container to finish..."
+          docker wait bootstrapper_l2
+          echo "Running transfer scripts..."
+          cd deps/scripts/transfer_from_L1
+          npm install
+          npm run transfer-l1
+          kill $MADARA_PID
```

Comment:
DRY: wrap this code into one or more scripts

Reply:
this can be done in a following PR

Reply:
`Setup app chain and run tests` is duplicated for mac and linux. Can we move this into a composite?